tools/themis/src/rules.rs

```diff
@@ -402,3 +402,52 @@ pub fn recursively_publishable(workspace: &Workspace) -> anyhow::Result<()> {
     }
     Ok(())
 }
+
+/// Ensure that top-level `Cargo.toml` does not contain superfluouos dependencies.
+pub fn no_superfluous_deps(workspace: &Workspace) -> anyhow::Result<()> {
+    let mut workspace_deps = BTreeSet::<String>::new();
+    let mut read_deps = |manifest: &toml::value::Value, table_name: &'static str| {
```

Comment:
nit: consider making it a helper function

Reply:
This closure captures `workspace_deps` so it cannot be a free-standing function.

Reply:
Can you pass it as a mutable arg?

Reply:
I mean yeah, but why? The whole point of a closure is to abstract away passing arguments in every call site. And its not like making this a free-standing function will make the code much cleaner. The only thing a free standing function resolves is releasing a mutable borrow on `workspace_deps` in between calls of the function, but given relatively straightforward situation with the lifetimes here, that also doesn’t seem like a strong motivation.

Reply:
Sure, I'm totally fine with it if this is your preferred code style.

Reply:
Thank you for the review!